src/metal.cs

```diff
@@ -3388,16 +3881,330 @@ interface MTLIndirectCommandBuffer : MTLResource {
 		[Abstract]
 		[Export ("indirectRenderCommandAtIndex:")]
 		IMTLIndirectRenderCommand GetCommand (nuint commandIndex);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[TV (13,0), NoMac, iOS (13,0)]
+		[Export ("indirectComputeCommandAtIndex:")]
+		IMTLIndirectComputeCommand GetIndirectComputeCommand (nuint commandIndex);
 	}
 
-	[NoiOS, NoTV, Mac (10,14)]
+	[iOS (13,0), TV (13,0), Mac (10,14)]
 	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
 	interface MTLSharedTextureHandle : NSSecureCoding {
 		[Export ("device")]
 		IMTLDevice Device { get; }
 
 		[NullAllowed, Export ("label")]
 		string Label { get; }
 	}
+
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoMac, NoTV, iOS (13,0)]
+	[Advice ("This API is not available when using UIKit on macOS.")]
+	[BaseType (typeof(NSObject))]
+	interface MTLRasterizationRateSampleArray {
+		[Export ("objectAtIndexedSubscript:")]
+		NSNumber Get (nuint index);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void Set (NSNumber value, nuint index);
```

Comment:
we have not been very consistent on similar API (inside Metal) but this would be different again
Use `GetObject` and `SetObject` for the names (better than the other alternative)